### PR TITLE
Float: Introduce floating point api eq

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* The eq(equal) api(s) introduced.
* Add new type fp_bits_x2.
* Add new type fp_bool_and_flags.
* Overload operator == for test to compare fp_bool_and_flags tuple.
* Add test cases for half, single and double floating point.
* Add eq(equal) to interface.

### CODE REVIEW

**Comment by Incarnation-p-lee** [comment]
This PR may conflict with #565 , thus we need something like rebase after #565 merged.

**Comment by Alasdair** [comment]
I'm away today, but it shouldn't be a problem for me to do a rebase before merging when I'm back.

**Comment by Incarnation-p-lee** [comment]
> I'm away today, but it shouldn't be a problem for me to do a rebase before merging when I'm back.

No hurry and take your time. You can review #565 first and then I can do the rebase anyway. I am afraid that you may review these 2 PR(s) and then merge. Then you may need to resolve the conflict, which is my duty.

File: lib/float/common.sail [hunk 1]
```diff
@@ -74,4 +77,13 @@ function float_decompose(op) = {
   }
 }
 
+/* Floating point operator implementations */
+infix    2 ==_t /* The suffix _t indicates tuple */
```
Comment (arichardson) [inline lib/float/common.sail]: I think these are bad for readability and it would be better to just spell it out.

Comment (arichardson) [reply lib/float/common.sail]: Or just scoped to the test file.

Comment (Alasdair) [reply lib/float/common.sail]: Within a test file you could do `$include <generic_equality.sail>` and use `==`, but it shouldn't be included in ordinary library code.

Comment (Incarnation-p-lee) [reply lib/float/common.sail]: Thanks for comments, that make sense to me, will update it after #565 .

Comment (Incarnation-p-lee) [reply lib/float/common.sail]: Rebase and updated.

**Comment by Incarnation-p-lee** [comment]
Kindly ping for this PR, thanks!

**Pull request merged**